python/cuml/cuml/linear_model/ridge.pyx

```diff
@@ -377,3 +377,22 @@ class Ridge(UniversalBase,
 
     def get_attr_names(self):
         return ['intercept_', 'coef_', 'n_features_in_', 'feature_names_in_']
+
+    def _should_dispatch_cpu(self, func_name, *args, **kwargs):
+        """
+        Dispatch to CPU implementation when:
+        1. Multi-target regression is detected (y has more than 1 column or is 2D)
+        2. For fit-related functions only (fit, fit_transform, fit_predict)
+        """
+        if func_name == "fit" and len(args) > 1:
+            y_m, _, _, _ = input_to_cuml_array(args[1])
```

Comment:
Won't doing this here do an unnecessary host -> device transfer if `args[1]` is a host array? Is there a way for you to check the shape robustly enough without invoking a transfer?

Reply:
You are correct. I've fixed that.